### PR TITLE
fix(fe-fpm-writer): validate either "sap.fe.core" or "sap.fe.templates" library when creating a building block

### PR DESCRIPTION
**Pull request by AlinaGovoruhina**
Follow up PR for https://github.com/SAP/open-ux-tools/issues/2241

Throw error when both  "sap.fe.core" and "sap.fe.templates" libraries are missing when creating a building block.
Do not throw error if at least one library is present.
Adding missing libraries support will be added as a follow up.

### CODE REVIEW

File: packages/fe-fpm-writer/src/common/validate.ts [hunk 1]
```diff
@@ -58,3 +59,29 @@ export function validateBasePath(basePath: string, fs?: Editor, dependencies = [
...
 }
+
+/**
+ * Validates base path, checks and adds missing libraries.
+ *
+ * @param {string} basePath - the base path
+ * @param {Editor} fs - the memfs editor instance
+ * @param {boolean} addMissingLibraries - indicates whether to add missing libraries or throw an error
+ */
+export function checkRequiredLibraries(basePath: string, fs: Editor, addMissingLibraries = false): void {
```
Comment (ManuelPastorini) [inline packages/fe-fpm-writer/src/common/validate.ts:70]: "check" should just check and return a result
"add" should add
and maybe there could be a method "ensureNeededDependencies" or similar, that first calls check, and based on the result calls add.

**Review by 815are — COMMENTED** [review]
As we checked with @AlinaGovoruhina then it simpler to change existing method `validateBasePath` than writing new method. We just need to change how `validateBasePath` checks dependency - instead of checking if one of passed dependencies are missing, we would need check if all passed dependencies are missing -> `OR` vs `AND`.
As we checked - in all current cases only single dependency is passed from outside and there was no case when multiple dependencies are passed for `validateBasePath`.
As conclusion, proposal - we just modify method `validateBasePath` instead of creating new method

File: packages/fe-fpm-writer/test/unit/building-block.test.ts [hunk 2]
```diff
@@ -72,10 +72,8 @@ describe('Building Blocks', () => {
         ).toThrowError(/Invalid view path/);
     });
 
-    test('validate sap.fe.core manifest dependency', async () => {
+    test('validate sap.fe.templates and sap.fe.core manifest dependencies are missing', async () => {
         const basePath = join(testAppPath, 'validate-manifest-dep');
-        fs.write(join(basePath, manifestFilePath), JSON.stringify(testManifestContent));
-
         // Test generator without sap.fe.core as dependency in manifest.json
         fs.write(join(basePath, manifestFilePath), JSON.stringify({ ...testManifestContent, 'sap.ui5': {} }));
```
Comment (815are) [inline packages/fe-fpm-writer/test/unit/building-block.test.ts:78]: can we please add additional test case when we have `sap.fe.templates` and `sap.fe.core` is missing? 
Like:
```
"sap.ui5": {
        "dependencies": {
            "libs": {
                "sap.fe.templates": {}
            }
        }
    }
```

Other tests for building block uses setup:
```
"sap.ui5": {
        "dependencies": {
            "libs": {
                "sap.fe.core": {}
            }
        }
    }
```

File: packages/fe-fpm-writer/src/common/validate.ts (hunk 1 above)
Comment (hitesh-parmar) [reply packages/fe-fpm-writer/src/common/validate.ts]: Reviewed by Andis https://github.com/SAP/open-ux-tools/pull/2284#pullrequestreview-2265845394

**Review by 815are — APPROVED** [review]
checked:
1. code changes are straightforward. Was not sure if we can change `validateBasePath` method logic - logic regarding how it validates multiple dependencies, but as we checked how method is called- we did not find any existing place where multiple dependencies are passed to `validateBasePath`. In result we are updating `validateBasePath` instead of creating new function. Method checks if at least one of passed dependency exists, then it considers as valid.
2. New tests are added;
3. Changelog persists

**Review by devinea — APPROVED** [review]

**Pull request merged**